src/Disciplina/DB/DSL/Interpret/Sqlite3.hs

```diff
@@ -0,0 +1,79 @@
+
+module Disciplina.DB.DSL.Interpret.Sqlite3 where
+
+import Universum
+
+import Control.Monad.Reader
+import Control.Monad.Catch
+import Control.Monad.Trans
+
+import qualified Data.Set as Set (Set, singleton, member)
+import Database.SQLite.Simple
+
+import Disciplina.DB.DSL.Class
+
+newtype Sqlite3T m a = Sqlite3T { getSqlite3T :: ReaderT Connection m a }
+    deriving (MonadTrans)
```

Comment:
You don't need this particular newtype because we'll have only one `ReaderT` over IO in our work monad. What you should have is:
- some typeclass `MonadSql` which can `query` and `execute` queries or something
- an `instance (HasLens' Connection ctx, MonadIO m) => ReaderT ctx m`, which performs these queries using a `Connection` from a context.
This way you'll automatically provide SQL capabilities to any working monad which will have opened Sqlite connection in its context (which is what we need).